packages/babel-plugin-transform-es2015-block-scoping/src/index.js

```diff
@@ -261,6 +261,12 @@ const loopVisitor = {
     const { node, parent, scope } = path;
     if (node[this.LOOP_IGNORE]) return;
 
+    if (node.trailingComments) {
+      node.trailingComments.forEach(comment => {
+        comment.value = comment.value.replace(/\n/g, "");
+      });
+    }
+
```

Comment:
For this to function properly, you need to locally replace the `babel-generator` lib's `index.js` with the lib built with the code in this branch